src/main/scala/zio/prelude/Associative.scala

```diff
@@ -1272,4 +1272,16 @@ trait AssociativeSyntax {
     def combine(r: => A)(implicit associative: Associative[A]): A =
       associative.combine(l, r)
   }
+
+  /**
+   * Provides syntax for combining values in a `ParSeq` which is thus done in a parallel manner.
+   */
+  implicit class AssociativeParSeqOps[A](v: collection.parallel.immutable.ParSeq[A]) {
+
+    /**
+     * Associatively combines the values in a parallel manner.
+     */
+    def reduceAssociative(implicit associative: Associative[A]): A =
```

Comment:
How about `reducePar`?

Reply:
I assume people will very often use it with `.par` right in front of the reduction, like `someSeqCollection.par.reduceAssociative`.
If that that is the case, wouldn't you be worried that `someSeqCollection.par.reducePar` could feel a bit repetitive on the "par" part? :)

I don't really mind, so if you think it's a good idea, let's do this :+1:

Reply:
How about just `reduce`, then?

Reply:
That method is already present "natively" in `ParIterable`, so I would be worried that could cause some confusion/ambiguity for our users, or won't work outright.
What do you think?
(I've reworked the code a bit, so you might want to have another look.)

Reply:
I've just tried renaming `IdentityParIterableOps#reduceIdentityBlocking` to just `reduce` and compilation in `ParIterable non-empty returns a value` fails with
```scala
[error] /home/ondra/Projects/zio-prelude/src/test/scala/zio/prelude/IdentitySpec.scala:63:61: missing argument list for method reduce in trait ParIterableLike
[error] Unapplied methods are only converted to functions when a function type is expected.
[error] You can make this conversion explicit by writing `reduce _` or `reduce(_)` instead of `reduce`.
[error]             assert(List(Sum(1), Sum(2), Sum(3), Sum(4)).par.reduce)(equalTo(Sum(10)))
[error]                                                             ^
[error] one error found
```